snorkel/candidates.py

```diff
@@ -88,12 +88,14 @@ def apply(self, context, clear, split, **kwargs):
                 bi, b = args[1]
 
                 # Check for self-joins, "nested" joins (joins from span to its subspan), and flipped duplicate
-                # "symmetric" relations
+                # "symmetric" relations. For symmetric relations, if mentions are of the same type, maintain
+                # their order in the sentence.
                 if not self.self_relations and a == b:
                     continue
                 elif not self.nested_relations and (a in b or b in a):
                     continue
-                elif not self.symmetric_relations and (b, a) in extracted:
+                elif not self.symmetric_relations and ((b, a) in extracted or
+                    (self.matchers[0] == self.matchers[1] and a.char_start > b.char_start)):
```

Comment:
I'm not sure that this `self.matchers[0] == self.matchers[1]` check will always work... but this seems fine to add in either way!